### PR TITLE
[POC WIP]benchmark for types package

### PR DESCRIPTION
**Pull request by SamYuan1990**
1. bench mark for types package
2. related with #656 enhance performance for type package.
- duplicate fetch from map
- init function missing
- useless memory allocated 
- try to build up no error return functions as we just logging error but do nothing else?

### CODE REVIEW

**Comment by SamYuan1990** [comment]
```
yuanyi@yuanyideMacBook-Pro types % go test -cpuprofile cpu.prof -memprofile mem.prof -bench . --count=5
goos: darwin
goarch: amd64
pkg: github.com/sustainable-computing-io/kepler/pkg/collector/metric/types
cpu: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
BenchmarkUInt64StatCollectionSetAggrStatMissed-16                         601155              2495 ns/op             183 B/op          3 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatMissed-16                         605821              2152 ns/op             182 B/op          3 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatMissed-16                         582312              2134 ns/op             186 B/op          3 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatMissed-16                         555684              2157 ns/op             191 B/op          3 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatMissed-16                         579454              2132 ns/op             186 B/op          3 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatBothRandom-16                     579511              2148 ns/op             186 B/op          3 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatBothRandom-16                     605644              2039 ns/op             182 B/op          3 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatBothRandom-16                     612538              2138 ns/op             181 B/op          3 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatBothRandom-16                     586896              2086 ns/op             185 B/op          3 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatBothRandom-16                     605199              2090 ns/op             182 B/op          3 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatCachedRandomNumber-16           62283670                17.19 ns/op            0 B/op          0 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatCachedRandomNumber-16           69439219                17.15 ns/op            0 B/op          0 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatCachedRandomNumber-16           59426810                18.42 ns/op            0 B/op          0 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatCachedRandomNumber-16           56135504                18.49 ns/op            0 B/op          0 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatCachedRandomNumber-16           62615835                18.28 ns/op            0 B/op          0 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatCached-16                       259816286                4.350 ns/op           0 B/op          0 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatCached-16                       278607254                4.421 ns/op           0 B/op          0 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatCached-16                       268708818                4.572 ns/op           0 B/op          0 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatCached-16                       262772254                4.412 ns/op           0 B/op          0 allocs/op
BenchmarkUInt64StatCollectionSetAggrStatCached-16                       260896773                4.401 ns/op           0 B/op          0 allocs/op
PASS
ok      github.com/sustainable-computing-io/kepler/pkg/collector/metric/types   28.293s
```

File: pkg/collector/metric/types/types.go [hunk 1]
```diff
@@ -53,6 +53,33 @@ func (s *UInt64Stat) SetNewDeltaValue(newDelta uint64, sum bool) error {
...
+func InitNewAggr(newAggr uint64) *UInt64Stat {
+	// verify aggregated value overflow
+	if newAggr == math.MaxUint64 {
+		return &UInt64Stat{
+			Aggr: 0,
+		}
+	} else {
+		return &UInt64Stat{
+			Aggr:  newAggr,
+			Delta: newAggr,
```
Comment (marceloamaral) [inline pkg/collector/metric/types/types.go:65]: The first Delta should be 0

**Review by marceloamaral — CHANGES_REQUESTED** [review]
Looks good. Just have few minor comments.

The test file should have the same name of the tested package.
`pkg/collector/metric/types/benchmark_test.go` -> `pkg/collector/metric/types/types_test.go`

**Comment by marceloamaral** [comment]
How is the profiling before and after this PR?

**Comment by SamYuan1990** [comment]
@marceloamaral I am going to split this PR into two, one for test case and another for code change.
in short the reduce time for a map fetch, which relays on contents in the map.

**Pull request closed without merging**